pkg/storage/ingest/reader.go

```diff
@@ -402,44 +428,39 @@ func (r *PartitionReader) fetchLastCommittedOffsetWithRetries(ctx context.Contex
 		err = retry.Err()
 	}
 
-	return offset, err
+	return 0, false, err
 }
 
 // fetchLastCommittedOffset returns the last consumed offset which has been committed by the PartitionReader
-// to the consumer group. If there is no offset committed, this function returns kafkaStartOffset constant,
-// which is a special value used to signal that partition should be consumed from the start.
-func (r *PartitionReader) fetchLastCommittedOffset(ctx context.Context) (int64, error) {
+// to the consumer group.
+func (r *PartitionReader) fetchLastCommittedOffset(ctx context.Context) (int64, bool, error) {
```

Comment:
New `bool` return value is not documented. We can either add documentation or name the return value to explain its meaning.

Reply:
Good point. Done in https://github.com/grafana/mimir/pull/7685/commits/79f629d77c97c97721242a3fae9d441eff6c2598